kliff/dataset/dataset.py

```diff
@@ -394,11 +732,220 @@ def _read(path: Path, weight: Optional[Weight] = None, file_format: str = "xyz")
             raise DatasetError(
                 f"No dataset file with file format `{file_format}` found at {parent}."
             )
+        return configs
 
-        logger.info(f"{len(configs)} configurations read from {path}")
+    def add_from_path(
+        self,
+        path: Path,
+        weight: Optional[Weight] = None,
+        file_format: str = "xyz",
+    ):
+        """
+        Read configurations from path and append them to dataset.
 
+        Args:
+            path: Path the directory (or filename) storing the configurations.
+            weight: an instance that computes the weight of the configuration in the loss
+                function.
+            file_format: Format of the file that stores the configuration, e.g. `xyz`.
+        """
+        configs = self._read_from_path(path, weight, file_format)
+        self.configs.extend(configs)
+
+    @classmethod
+    def from_ase(
+        cls,
+        path: Path = None,
+        ase_atoms_list: List[ase.Atoms] = None,
+        weight: Optional[Weight] = None,
+        energy_key: str = "energy",
+        forces_key: str = "forces",
+        slices: str = ":",
+        file_format: str = "xyz",
+    ) -> "Dataset":
+        """
+        Read configurations from ase.Atoms object and initialize a dataset. If the
```

Comment:
More explanation should be added on how to use it, particularly on the fact that a user is expected to provide either `path` or `ase_atoms_list`. Same for the `add_from_ase` function.

Reply:
Added example and more explanation in docstring